### ISSUE TITLE
Show lines between siblings in the inspector tree view.

### ISSUE THREAD

**Issue by jacob314** [flutter/flutter-intellij#1608]
Siblings might become too far away from one other if one’s child tree is big and fully expanded. Show dotted lines for offstage siblings, solid lines for onstage siblings matching the existing toStringDeep implementation and metadata that is already exposed by DiagnosticsNode.

**Comment by pq** [comment]
@alexander-doroshko: do you happen to know if there are hooks for drawing on the tree view if we wanted to render some extra lines to show these relationships?

**Comment by alexander-doroshko** [comment]
nothing off the top of my head, sorry

**Comment by scheglov** [comment]
In theory `BasicTreeUI` can paint lines with `paintHorizontalPartOfLeg` and `paintVerticalPartOfLeg`. But practically it actively resists. 

1. There is `paintLines = UIManager.getBoolean("Tree.paintLines")` in `BasicTreeUI`, and `intellijlaf.properties` sets it to `false`.

2. `WideSelectionTreeUI.shouldPaintLines()` has strict checks.

```java
    private boolean shouldPaintLines() {
        if (!UIUtil.isUnderAquaBasedLookAndFeel() && !UIUtil.isUnderDarcula() && !UIUtil.isUnderIntelliJLaF()) {
            return this.myForceDontPaintLines || !"None".equals(this.tree.getClientProperty("JTree.lineStyle"));
        } else {
            return false;
        }
    }
```

So, I suspect that we either will need to use some reflection, or paint ourselves.

**Comment by jacob314** [comment]
I've got this implemented doing it ourselves as the line style does not
play nicely with the modern icons indicating expand and collapse state.
I'll send you a CL for the tree rendering changes as you've also been
looking at this code.
I've also optimized as we only really care about showing lines for nodes
with multiple children and we care about showing property children
differently than other children (for views showing both properties and
children inline.
